### PR TITLE
JP-3533 Update emicorr to run on FAST mode MIRI data

### LINKED ISSUE #8264: Update emicorr to run on FAST mode MIRI data
_Issue [JP-3533](https://jira.stsci.edu/browse/JP-3533) was created on JIRA by [Misty Cracraft](https://jira.stsci.edu/secure/ViewProfile.jspa?name=cracraft):_

When testing the shower code with emicorr updates in build 10.1, I found a test case of a coronagraphic image that was still overly flagged (most of the image) as NaNs in the rate file. This file was in FAST mode. The emicorr code was only set up to run on FASTR1 and SLOWR1 data, which is the bulk of MIRI data, but TA data is taken in FAST mode and also needs to be corrected for emi, so that the MIRI team can properly calibrate coron data.

And we cannot turn on the shower code for coron data until the emicorr code properly treats FAST mode data.

We are not requesting that emicorr runs on FASTGRPAVG data, as Eddie pointed out that the group averaging beats down the noise and it would probably be more work to figure out how to correct it than the actual correction would benefit the data.

### PR DESCRIPTION
**Pull request by penaguerrero**
Resolves [JP-3533](https://jira.stsci.edu/browse/JP-3533)

Closes #8264 

This PR addresses the behavior of the emicorr step. Previously it would run always for all MIRI data. The MIRI team requested it is skipped when no reference file is found and to add a parameter to allow the user to run the step for given frequencies with an on-the-fly generated reference file.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by hbushouse** [comment]
@penaguerrero  Is this still WIP or is it ready for review?

**Comment by penaguerrero** [comment]
From my side it is ready but I was waiting to hear confirmation from Misty that it is indeed what they want.

File: jwst/emicorr/emicorr_step.py [hunk 1]
```diff
@@ -22,44 +22,56 @@ class EmiCorrStep(Step):
         nints_to_phase = integer(default=None)  # Number of integrations to phase
         nbins = integer(default=None)  # Number of bins in one phased wave
         scale_reference = boolean(default=True)  # If True, the reference wavelength will be scaled to the data's phase amplitude
-        skip = boolean(default=True)
+        skip = boolean(default=True)  # Skip the step
+        onthefly_corr_freq = float_list(default=None)  # Frequencies number to use for correction
```
Comment (hbushouse) [inline jwst/emicorr/emicorr_step.py:26]:
```suggestion
        onthefly_corr_freq = float_list(default=None)  # Frequencies to use for correction
```

**Comment by hbushouse** [comment]
Regression test run?

**Comment by penaguerrero** [comment]
regression test: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1308/

**Review by hbushouse — APPROVED** [review]
Looks good.

**Pull request merged**